client/src/pages/result/Result.js

```diff
@@ -1,116 +1,197 @@
 import "./Result.css";
 import Form from "react-bootstrap/Form";
 import { useSearchParams } from "react-router-dom";
-import Stack from "react-bootstrap/Stack";
+import SendSmsButton from "./button/sendSMSButton";
+import SendEmailButton from "./button/sendEmailButton";
+import LocationIcon from "../../components/LocationIcon";
 
 export default function Result() {
 	let [searchParams, setSearchParams] = useSearchParams();
 	const [service, setService] = useState(searchParams.get("service"));
 	const [location, setLocation] = useState(searchParams.get("location"));
 	const [data, setData] = useState([]);
+	const [emailSent, setEmailSent] = useState(false);
+	const [smsSent, setSmsSent] = useState(false);
 	useEffect(() => {
 		fetch(`/api/ngo?service=${service}&location=${location}`)
 			.then((response) => response.json())
 			.then((data) => {
-        console.log(data);
-        setData(data);
-}
-        );
+				console.log(data);
+				setData(data);
+			});
 	}, [service, location]);
 
 	function selectService(e) {
-		setService(e.target.value);
-		setSearchParams({
-			service: e.target.value,
-			location: location,
-		});
+		if (e.target.checked) {
+			setService(e.target.id);
+			setSearchParams({
+				service: e.target.id,
+				location: location,
+			});
+		}
 	}
 
 	function selectLocation(e) {
-		setLocation(e.target.value);
-		setSearchParams({
-			service: service,
-			location: e.target.value,
-		});
+		if (e.target.checked) {
+			setLocation(e.target.id);
+			setSearchParams({
+				service: service,
+				location: e.target.id,
+			});
+		}
+	}
+	function sendEmail(email) {
+		if (!email) {
+			return;
+		}
+		fetch("/api/sendmail", {
+			method: "POST",
+			headers: {
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify({
+				to: email,
+				data: data,
+				service: service,
+				location: location,
+			}),
+		})
+			.then((response) => {
+				if (response.ok) {
+					setEmailSent(true);
+				} else {
+					throw new Error("Failed to send email");
+				}
+			})
+			.catch((error) => console.error(error));
+	}
+	function sendSms(sms) {
+		if (!sms) {
+			return;
+		}
+		fetch("/api/sendsms", {
+			method: "POST",
+			headers: {
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify({
+				to: sms,
+				data: data,
+				service: service,
+				location: location,
+			}),
+		})
+			.then((response) => {
+				if (response.ok) {
+					setSmsSent(true);
+				} else {
+					throw new Error("Failed to send SMS");
+				}
+			})
+			.catch((error) => console.error(error));
 	}
 	return (
-		<>
-			<Form>
-				<Form.Group className="select-group" controlId="service">
-					<Form.Label>Service</Form.Label>
-					<Form.Select
-						aria-label="Service"
-						value={service}
-						onChange={selectService}
+		<div className="d-flex">
+			<div className="col-3 bg-light rounded m-2">
+				<Form>
+					<Form.Group
+						className="select-group"
+						style={{ color: "#004e87" }}
```

Comment:
Actually, this color needs to be re-defined if we use the Bootstrap class, this color code is not actually the same primary color of Bootstrap, in order to re-define Bootstrap variables I need to understand it, but I still don't understand how to do it

Reply:
Yes that color is used due to our style guide, and it is customizable but this method should be used for the primary purpose I mean if we want to affect our main project by changing variables we can create a stylesheet like custom.scss and afterward importing and re-defining the variables there will be integration with the main.css file and after that, all variables will be inserted to our main.css file I tried that and there are hundreds of new lines appearing for only changing an inline style that actually may affect the rest of the project, therefore my opinion is continuing with this way instead of customizing variables